From clemens@ladisch.de  Fri Apr 24 15:32:40 2009
From: Clemens Ladisch <clemens@ladisch.de>
Date: Fri, 24 Apr 2009 10:11:40 +0200
Subject: nls: utf8_wcstombs: use correct buffer size in error case
To: Greg KH <greg@kroah.com>
Cc: Eugen Dedu <Eugen.Dedu@pu-pm.univ-fcomte.fr>, Alan Stern <stern@rowland.harvard.edu>
Message-ID: <49F1743C.3030705@ladisch.de>


When utf8_wcstombs encounters a character that cannot be encoded, we
must not decrease the remaining output buffer size because nothing has
been written to the output buffer.

Signed-off-by: Clemens Ladisch <clemens@ladisch.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 fs/nls/nls_base.c |    1 -
 1 file changed, 1 deletion(-)

--- a/fs/nls/nls_base.c
+++ b/fs/nls/nls_base.c
@@ -144,7 +144,6 @@ utf8_wcstombs(__u8 *s, const wchar_t *pw
 			size = utf8_wctomb(op, *ip, maxlen);
 			if (size == -1) {
 				/* Ignore character and move on */
-				maxlen--;
 			} else {
 				op += size;
 				maxlen -= size;
